README.md

```diff
@@ -64,7 +64,9 @@ The implementation is covered by the protocol buffers
 
 ### TypeScript
 
-The minimum version of TypeScript supported by Protobuf-ES is **4.1.2**.
+The minimum version of TypeScript supported by Protobuf-ES is **4.1.2**.  Our compatibility tests use the default 
+settings for `tsconfig.json` that are generated by running `tsc --init` (with a few minor exceptions).  As a result,
+for any version we support, we inherently support the default settings respective to that version of TypeScript.
```

Comment:
What are the exceptions?

Reply:
We specify `skipLibCheck: false` down to 4.4.4. and `moduleResolution: NodeNext` where it is supported.

Reply:
Yes, but both settings are actually _more_ restrictive than the defaults. So we are compatible with the default compiler settings for all versions of TypeScript >= 4.1.2.

I feel like I have to read the paragraph carefully, perhaps more than once, to understand what's supported. 

Isn't this simple alternative also true?

> The generated code is compatible with TypeScript v4.1.2 or later, with the default compiler settings.

Reply:
Sure.  I just wanted to be clear in case anyone noticed that we don't officially use the defaults in all versions.

Reply:
We already have comments in the tsconfig files that _should_ explain that. If it isn't clear enough, let's improve them. Or move the more restrictive settings to separate tsconfig files, or remove them completely.

Reply:
Sounds good.  Updated.

Reply:
TY. I only had reservations about the top-level readme - the details you added in https://github.com/bufbuild/protobuf-es/pull/319/commits/467154d3e97d77a982c6fa05bd113b3fb720f542#diff-c73cc5d535f51338cff7c9594228cae1b32f902aeeac134be925efc2a7c9af35 seem helpful!

Regarding strictNullChecks - protobuf-ts has a [note in it's manual](https://github.com/timostamm/protobuf-ts/blob/master/MANUAL.md#oneof-representation). Considering a user ran into the issue, perhaps we should add a similar note to [the docs here](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md#oneof-groups)? We no longer have to explain that users need to set `strict` to true, it's all on by default in all TS version we support. But it'll be helpful for users who have strictNullChecks disabled for legacy reasons and run into this.

Reply:
Oh nice.  OK, I added that note in and put my coment on the test README back in.